crates/sui-graphql-rpc/src/types/transaction_block/mod.rs

```diff
@@ -598,3 +509,146 @@ impl TryFrom<TransactionBlockEffects> for TransactionBlock {
         })
     }
 }
+
+fn apply_scan_limited_pagination(
+    conn: &mut ScanConnection<String, TransactionBlock>,
+    page: &Page<Cursor>,
+    transactions: &Vec<StoredTransaction>,
+    tx_bounds: TxBounds,
+    scan_limit: Option<u64>,
+    checkpoint_viewed_at: u64,
+) {
+    if page.is_from_front() {
+        apply_forward_scan_limited_pagination(
+            conn,
+            page,
+            transactions,
+            tx_bounds,
+            scan_limit,
+            checkpoint_viewed_at,
+        );
+    } else {
+        apply_backward_scan_limited_pagination(
+            conn,
+            page,
+            transactions,
+            tx_bounds,
+            scan_limit,
+            checkpoint_viewed_at,
+        );
+    }
+}
+
+/// In the forward scan-limited pagination scenario, there are two special cases when dealing with
```

Comment:
It would be nice to map these comments to the parts of the implementation that deal with the various cases (similar to how `paginate_results` is written).